src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxResolvingClassLoader.java

```diff
@@ -72,42 +57,29 @@
     }
 
     @Override public URL getResource(String name) {
-        ClassLoader parentLoader = getParent();
-        return load(parentResourceCache, name, parentLoader, () -> Optional.fromNullable(parentLoader.getResource(name))).orNull();
+        return parentResourceCache.get(getParent()).getUnchecked(name).orNull();
     }
 
-    // We cannot have the inner cache be a LoadingCache and just use .get(name), since then the values of the outer cache would strongly refer to the keys.
-    private static <T> T load(LoadingCache<ClassLoader, Cache<String, T>> cache, String name, ClassLoader parentLoader, Supplier<T> supplier) {
-        try {
-            return cache.getUnchecked(parentLoader).get(name, () -> {
-                Thread t = Thread.currentThread();
-                String origName = t.getName();
-                t.setName(origName + " loading " + name);
-                long start = System.nanoTime(); // http://stackoverflow.com/q/19052316/12916
-                try {
-                    return supplier.get();
-                } finally {
-                    t.setName(origName);
-                    long ms = (System.nanoTime() - start) / 1000000;
-                    if (ms > 1000) {
-                        LOGGER.log(Level.INFO, "took {0}ms to load/not load {1} from {2}", new Object[] {ms, name, parentLoader});
+    private static <T> ClassLoaderCache<LoadingCache<String, T>> makeParentCache(BiFunction<ClassLoader, String, T> func) {
+        // TODO expire cache if new plugins are installed
+        return new ClassLoaderCache<>(parentLoader -> {
+            return CacheBuilder.newBuilder().build(new CacheLoader<String, T>() {
```

Comment:
Not using anything interesting from Guava here; could just replace with, say, `ConcurrentHashMap.computeIfAbsent`.